projects/plugins/jetpack/_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -990,6 +990,46 @@ public function update_data( $request ) {
 					$updated = (bool) get_option( $option ) !== (bool) $value ? update_option( $option, (bool) $value ) : true;
 					break;
 
+				case 'subscription_options':
+					if ( ! is_array( $value ) ) {
+						break;
+					}
+
+					$allowed_keys   = array( 'invitation', 'comment_follow', 'welcome' );
+					$filtered_value = array_filter(
+						$value,
+						function ( $key ) use ( $allowed_keys ) {
+							return in_array( $key, $allowed_keys, true );
+						},
+						ARRAY_FILTER_USE_KEY
+					);
+
+					if ( empty( $filtered_value ) ) {
+						break;
+					}
+
+					array_walk_recursive(
+						$filtered_value,
+						function ( &$value ) {
+							$value = wp_kses(
+								$value,
+								array(
+									'a' => array(
+										'href' => array(),
+									),
+								)
+							);
+						}
+					);
+
+					$old_subscription_options = get_option( 'subscription_options' );
+					$new_subscription_options = array_merge( $old_subscription_options, $filtered_value );
```

Comment:
Looks like this returns a boolean when the option has not been set before, causing the `array_merge()` to fail.

Reply:
Good point! Putting a check for `old_subscription_options` to be array:
```
$old_subscription_options = get_option( 'subscription_options' );
if ( ! is_array( $old_subscription_options ) ) {
	$old_subscription_options = array();
}
$new_subscription_options = array_merge( $old_subscription_options, $filtered_value );
```

Reply:
Tested and thsi now works.